balancer-js/src/modules/contracts/contracts.module.ts

```diff
@@ -1,78 +1,96 @@
-import { EulerLinearPoolFactory } from '@/modules/contracts/implementations/factories/euler-linear-pool-factory';
-import { YearnLinearPoolFactory } from '@/modules/contracts/implementations/factories/yearn-linear-pool-factory';
-import { GearboxLinearPoolFactory } from '@/modules/contracts/implementations/factories/gearbox-linear-pool-factory';
+import {
+  AaveLinearPoolFactory,
+  AaveLinearPoolFactory__factory,
+  BalancerHelpers,
+  BalancerHelpers__factory,
+  ComposableStablePoolFactory,
+  ComposableStablePoolFactory__factory,
+  ERC20,
+  ERC20__factory,
+  ERC4626LinearPoolFactory,
+  ERC4626LinearPoolFactory__factory,
+  EulerLinearPoolFactory,
+  EulerLinearPoolFactory__factory,
+  GaugeClaimHelper,
+  GaugeClaimHelper__factory,
+  GearboxLinearPoolFactory,
+  GearboxLinearPoolFactory__factory,
+  LidoRelayer,
+  LidoRelayer__factory,
+  LiquidityGaugeV5,
+  LiquidityGaugeV5__factory,
+  RelayerV3,
+  RelayerV3__factory,
+  RelayerV4,
+  RelayerV4__factory,
+  RelayerV5,
+  RelayerV5__factory,
+  Vault,
+  Vault__factory,
+  WeightedPoolFactory,
+  WeightedPoolFactory__factory,
+  YearnLinearPoolFactory,
+  YearnLinearPoolFactory__factory,
+} from '@/contracts';
+import { Network } from '@/lib/constants/network';
+import { BALANCER_NETWORK_CONFIG } from '@/lib/constants/config';
+import { ContractAddresses } from '@/types';
 
 type ContractFactory = (
   address: string,
   signerOrProvider: Signer | Provider
 ) => Contract;
 
 export interface ContractInstances {
-  aaveLinearPoolFactory?: Contract;
+  aaveLinearPoolFactory?: AaveLinearPoolFactory;
   balancerHelpers: BalancerHelpers;
   BasePool: ContractFactory;
-  composableStablePoolFactory?: Contract;
+  composableStablePoolFactory?: ComposableStablePoolFactory;
   ERC20: ContractFactory;
-  erc4626LinearPoolFactory?: Contract;
-  eulerLinearPoolFactory?: Contract;
-  gaugeClaimHelper?: Contract;
-  gearboxLinearPoolFactory?: Contract;
+  erc4626LinearPoolFactory?: ERC4626LinearPoolFactory;
+  eulerLinearPoolFactory?: EulerLinearPoolFactory;
+  gaugeClaimHelper?: GaugeClaimHelper;
+  gearboxLinearPoolFactory?: GearboxLinearPoolFactory;
   lidoRelayer?: LidoRelayer;
   liquidityGauge: ContractFactory;
   multicall: Contract;
-  relayerV3?: Contract;
-  relayerV4?: Contract;
-  relayerV5?: Contract;
+  relayerV3?: RelayerV3;
+  relayerV4?: RelayerV4;
+  relayerV5?: RelayerV5;
   vault: Vault;
   veBal?: VeBal;
   veBalProxy?: VeBalProxy;
-  weightedPoolFactory?: Contract;
-  yearnLinearPoolFactory?: Contract;
+  weightedPoolFactory?: WeightedPoolFactory;
+  yearnLinearPoolFactory?: YearnLinearPoolFactory;
 }
 
 export class Contracts {
-  aaveLinearPoolFactory?: Contract;
-  balancerHelpers: BalancerHelpers;
-  composableStablePoolFactory?: Contract;
+  aaveLinearPoolFactory?: ContractInstances['aaveLinearPoolFactory'];
+  balancerHelpers: ContractInstances['balancerHelpers'];
+  composableStablePoolFactory?: ContractInstances['composableStablePoolFactory'];
```

Comment:
I'm not super comfortable with declaring instances with strings like this - feels error prone.
And I believe we could also avoid this duplication.
Maybe group everything together under `instances: ContractInstances`?
What do you think?

Reply:
I think it's very valid to discuss this topic, to be honest, I don't feel like the errors caused by this type of declaration are a problem because this type errors are accused immediately by the lint, in the same file, for example:
Ex.: Contracts and ContractsInstace have different parameters name, like "composableStablePoolFactory" and "composableStableFactory", if we reference by the wrong name, like this:

`composableStablePoolFactory: ComposableStablePoolFactory`
`...`
`composableStableFactory?: ContractInstances['composableStableFactory']`

The type of Contracts "composableStableFactory" is "never" instead of "ComposableStablePoolFactory" and this is accused by lint(in the line that executes ComposableStablePoolFactory__factory.connect" function). 

Now on the other hand, adding a `contractInstances` would not make any negative impact on the module's code or usability at all, since the contracts are accessed using `balancerContracts.contracts` get function, and would spare the task to declare each new contract two times, one in the ContractInstances and one in the Contracts. So, yeah, in my opinion it's a better option to use it, thanks!